File: golang/3-lengthOfLongestSubstring.go
package golang

import "bytes"

func LengthOfLongestSubstring(s string) int {
	temp := []byte{}
	max := 0
	for i := range s {
		if index := bytes.IndexByte(temp, s[i]); index > -1 {
			temp = temp[index+1:]
		}
		temp = append(temp, s[i])
		length := len(temp)
		if max < length {
			max = length
		}
	}
	return max
}

File: ts/3-lengthOfLongestSubstring.ts
export const lengthOfLongestSubstring = (s: string): number => {
  let temp = ''
  let max = 0

  for (let i = 0; i < s.length; i++) {
    const index = temp.indexOf(s[i])
    if (index !== -1) {
      temp = temp.substring(index + 1)
    }
    temp += s[i]
    const len = temp.length
    if (max < len) {
      max = len
    }
  }

  return max
}
